inc/Engine/License/API/User.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace WP_Rocket\Engine\License\API;
+
+class User {
+	/**
+	 * The user object
+	 *
+	 * @var object
+	 */
+	private $user;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param object $user The user object.
+	 */
+	public function __construct( $user ) {
+		$this->user = $user;
+	}
+
+	/**
+	 * Gets the user license type
+	 *
+	 * @return int
+	 */
+	public function get_license_type() {
+		if (
+			! is_object( $this->user )
+			||
+			! isset( $this->user->licence_account )
+		) {
+			return 0;
+		}
+
+		return (int) $this->user->licence_account;
+	}
+
+	/**
+	 * Gets the user license expiration timestamp
+	 *
+	 * @return int
+	 */
+	public function get_license_expiration() {
+		if (
+			! is_object( $this->user )
```

Comment:
As this check is repeating, how about if we create a private method to encapsulate the check? Then there's only 1 place where we need to change the code in the future (if needed).

Reply:
Or another idea: 

- Check that it's an object in the constructor. 
- If no, instantiate to `stdClass`. 
- Then remove all of the `is_object` checks.